### PR TITLE
Add select platform for mopping

### PR DESCRIPTION
**Pull request by Lash-L**
hey @humbertogontijo can you test this for me? 

I'm currently not at my home for a few days and my robot is dead/trapped during a cleaning.

I'm adding this for the core integration so I figured I would add it here, I think I copied over everything correctly.

But could you let me know if this works?  If you want you could test the actual core PR, but that isn't necessarily needed.

### CODE REVIEW

**Comment by Lash-L** [comment]
Hey @humbertogontijo I updated the core request with the change you said, if you have a chance can you test it again? 

I should be able to go back to developing and testing tomorrow.

File: custom_components/roborock/select.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+"""Support for Roborock select."""
+from dataclasses import dataclass
+from __future__ import annotations
```
Comment (humbertogontijo) [inline custom_components/roborock/select.py:3]:
```suggestion
"""Support for Roborock select."""
from __future__ import annotations

from dataclasses import dataclass
```

**Review by humbertogontijo — CHANGES_REQUESTED** [review]

**Comment by humbertogontijo** [comment]
I avoided creating a file named select.py for the project just because of a bug in PyCharm debugging(just running it, everything works).
Since we can't change the name of the file, do you have any idea how to solve this?
image

**Comment by Lash-L** [comment]
> I avoided creating a file named select.py for the project just because of a bug in PyCharm debugging(just running it, everything works). Since we can't change the name of the file, do you have any idea how to solve this? image

Hm i'm not sure. I doubt this would fix it, but potentially changing to using Platforms(Enum in home assistant const).SELECT instead of SELECT_DOMAIN may help?

**Comment by humbertogontijo** [comment]
Found a way to overcome it. Just changing the working directory when debugging something

**Comment by humbertogontijo** [comment]
Since your not home I'll do myself the changes I requested

**Comment by humbertogontijo** [comment]
About the core integration PR. I'd recommend bumping api version to latest. Yesterday I made huge improvements to the local_api.py

**Comment by Lash-L** [comment]
> About the core integration PR. I'd recommend bumping api version to latest. Yesterday I made huge improvements to the local_api.py

Yeah I noticed that! They typically like version bumps to be their own PR where that is all that is changed, so I'll give it a try.

Any breaking changes I should know about? I can't test it right now

**Comment by Lash-L** [comment]
@humbertogontijo I made some changes to core that should probably be copied over here.

I was incorrectly assigning unique_id and the select entities did not have a name - I'm surprised it worked with both of these problems.

**Comment by humbertogontijo** [comment]
Instead of using a while is_connected to receive socket messages(Which was terribe). Its using loop.create_connection

**Comment by humbertogontijo** [comment]
Btw, I had to make some changes to the selected entity for it to work
@Lash-L check it out https://github.com/humbertogontijo/homeassistant-roborock/commit/93b0d743b9f08d15ca2def8aea6d46311dfa650f

**Pull request merged**